pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -202,8 +202,37 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus) *kubecontainer.Runtim
 	return &kubecontainer.RuntimeStatus{Conditions: conditions}
 }
 
+func (m *kubeGenericRuntimeManager) getSeccompProfile(annotations map[string]string, containerName string,
+	podSecContext *v1.PodSecurityContext, containerSecContext *v1.SecurityContext) string {
+	fieldProfile := func(scmp *v1.SeccompProfile) string {
+		if scmp.Type == v1.SeccompProfileUnconfined {
+			return v1.SeccompProfileNameUnconfined
+		}
+		if scmp.Type == v1.SeccompProfileTypeDefault {
+			return v1.SeccompProfileRuntimeDefault
+		}
+		if scmp.Type == v1.SeccompProfileLocalhost && scmp.LocalhostProfile != nil {
+			fname := filepath.Join(m.seccompProfileRoot, *scmp.LocalhostProfile)
+			return v1.SeccompLocalhostProfileNamePrefix + fname
+		}
+
+		return ""
+	}
+
+	if containerSecContext != nil && containerSecContext.SeccompProfile != nil {
+		return fieldProfile(containerSecContext.SeccompProfile)
+	}
+
+	if podSecContext != nil && podSecContext.SeccompProfile != nil {
+		return fieldProfile(podSecContext.SeccompProfile)
+	}
+
+	return m.getSeccompProfileFromAnnotations(annotations, containerName)
```

Comment:
the KEP described a lookup path that did the following:

1. container-specific field
2. container-specific annotation
3. pod-level field
4. pod-level annotation

This implementation would honor a pod-level field over a container-level annotation

Reply:
I added an unit test to cover the missing scenario and corrected the priority order.